### PR TITLE
Add the `transformer_discriminative` schedule

### PR DESCRIPTION
**Pull request by danieldk**
## Description

This schedule wraps two schedules:

1. `transformer_schedule` is used for transformer parameters.
2. `default_schedule` is used for all other parameters.

This differentation allows you to implement good transformer finetuning practices such as:

- Initially freezing the transformer while freshly-initialized classifier parameter settle to avoid catastrophic forgetting in the transformer due to the initially large gradients.
- Using smaller learning rates for the transformer than classification layers.

### Types of change

Feature

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by shadeMe — APPROVED** [review]
LGTM (the tests need fixing, though)! One minor comment.

File: spacy_curated_transformers/schedules.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    default_schedule: Schedule,
+    transformer_schedule: Schedule,
+) -> Schedule:
+    """Discriminative learning rate schedule for transformer encoders.
+
+    This schedule uses `transformer_schedule` for all transformer encoder
+    parameters and `default_schedule` for other parameters.
+
+    default_schedule (Schedule): default schedule.
+    transformer_schedule (Schedule): schedule for transformer parmeters.
```
Comment (svlandeg) [inline spacy_curated_transformers/schedules.py:19]:
```suggestion
    transformer_schedule (Schedule): schedule for transformer parameters.
```

File: spacy_curated_transformers/tests/test_schedules.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+from spacy_curated_transformers.schedules import transformer_discriminative
+
+
+def test_schedules():
+    default_schedule = constant(1e-3)
+    transformer_schedule = constant(1e-5)
+    schedule = transformer_discriminative(default_schedule, transformer_schedule)
+    assert schedule(0, key=(0, "some_key")) == 1e-3
+    assert schedule(0, key=(1, "curated_encoder.embeddings")) == 1e-5
```
[the remark this answers is not in the document]
Comment (danieldk) [reply spacy_curated_transformers/tests/test_schedules.py:11]: Fixed, thanks.

(Took a while because a new spaCy 4/Thinc 9 dev releases were needed.)
